### ISSUE TITLE
Conversion PyTorch => ONNX => TensorRT

### ISSUE THREAD

**Issue by michieljanssen97** [qubvel/segmentation_models.pytorch#272]
Hi,

I'm trying to convert a segmentation model (ENCODER = efficientnet-b2, DECODER = FPN) to ONNX and afterwards to TensorRT (TRT).
Converting to ONNX seems to work but I can't get the conversion to TRT right. I tried the 'torch2trt' library but couldn't succeed...

Does anyone has experience with this?

Running:
efficientnet-pytorch==0.6.3
onnx==1.7.0
segmentation-models-pytorch==0.1.0
torch==1.6.0
torch2trt==0.1.0
torchvision==0.7.0

Thanks in advance,

Michiel

**Comment by eikaramba** [comment]
How did you convert to onnx?
I used 
```
x = torch.randn(1, 3, 256, 256)
best_model.to("cpu")
torch.onnx.export(best_model, x, 'model.onnx',opset_version=11)
```
however afterwards i can run it but the output is completely black, no objects are detected whatsoever

**Comment by michieljanssen97** [comment]
> How did you convert to onnx?
> I used
> 
> ```
> x = torch.randn(1, 3, 256, 256)
> best_model.to("cpu")
> torch.onnx.export(best_model, x, 'model.onnx',opset_version=11)
> ```
> 
> however afterwards i can run it but the output is completely black, no objects are detected whatsoever

**Comment by michieljanssen97** [comment]
model = torch.load('./model.pth')
model = model.cuda()
model.eval()

dummy_input = torch.randn(1, 3, 704, 1280, device='cuda')
torch.onnx.export(model, dummy_input, "model.onnx", input_names=["input"], output_names=["output"], verbose=True, opset_version=11)

**Comment by John-Trager** [comment]
Hey @eikaramba did you ever sort out why the output was blank for the onnx model?

**Comment by eikaramba** [comment]
@John-Trager no we use pytorch mobile optimization for now and directly pytorch on the phone. hopefully the mobile support of pytorch and especially hardware acceleration will improve in the future.

**Comment by a-beginner-debug** [comment]
After converting to the onnx model, use opencv's dnn module to call, and the output result is:
out: [[[[-31.546917 -33.87421  -36.201504 ... -49.26362  -46.150883
    -43.037766]
   [-33.46579  -35.773727 -38.081665 ... -54.56298  -50.830086
    -47.096737]
   [-35.38466  -37.67324  -39.961823 ... -59.862347 -55.509293
    -51.15571 ]
   ...
   [-36.907887 -39.365772 -41.823654 ... -29.387836 -27.53714
    -25.686216]
   [-33.64544  -35.990234 -38.335026 ... -27.419733 -25.870855
    -24.321785]
   [-30.38299  -32.614693 -34.846394 ... -25.451626 -24.204567
    -22.957355]]]]

the code is:
def inference_onnx(model_weight, img):

    net = cv2.dnn.readNetFromONNX(model_weight)  
    image = cv2.imread(img)  
    image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
    image=image.astype(np.float32, copy=False)
    PIXEL_MEANS=(0.485, 0.456, 0.406)
    PIXEL_STDS=(0.229, 0.224, 0.225)
    print("image.shape:", image.shape)
    image/=255.0
    image-=np.array(PIXEL_MEANS)
    image/=np.array(PIXEL_STDS)

    blob = cv2.dnn.blobFromImage(image, size=(1280, 704)) 
    net.setInput(blob) 
    out = net.forward("output")
    pr_mask = out.squeeze().numpy().round()

### Obviously the result is wrong,  who can find the problem with the result

**Comment by Fritskee** [comment]
Same issue here! 
converted the model to ONNX and the obtained results are just plain wrong! Has anybody figured out the issue to what this could be? @qubvel maybe?

**Comment by John-Trager** [comment]
Just to follow up to this I resolved my personal issue by normalizing the pixel values from 0-255 to 0-1 when converting the model using ONNX, which resulted in the model to work (as opposed to having a completely black output).

**Comment by neo133** [comment]
were you able to convert to TensorRT after ONNX? if so, how? @John-Trager

**Comment by davodogster** [comment]
@neo133 @michieljanssen97  Hi Were you able to convert SegModelsPytorchLightning model to ONNX to TensorRT?
